.github/workflows/main.yml

```diff
@@ -20,15 +20,17 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v1
-      - name: Setup node ${{ matrix.node_version }}
-        uses: actions/setup-node@v1
-        with:
-          node-version: ${{ matrix.node_version }}
 
       - uses: pnpm/action-setup@v2
         with:
           version: 8.5.1
 
+      - name: Setup node ${{ matrix.node_version }}
+        uses: actions/setup-node@v3
```

Comment:
We need a major update of setup-node to use cache option. But I think it's ok since it's already used in release.yml with same options.